### PR TITLE
React & Redux group project - Space Travelers' Hub

### PR DESCRIPTION
**Pull request by akos29**
### React & Redux group project - Space Travelers' Hub

In this request majorly:

- add setup of a react-redux project
- add fetch data and update the Redux store for Rockets and Missions
- add a feature to render the list from the store with style
- add the required actions and reducers for booking rockets and joining missions
- add actions and reducers for canceling rockets and missions
- add features for conditional components rendering
- add My Profile section to show booked rockets and joined missions
- add testing for components

### CODE REVIEW

File: src/routes/destroy.jsx [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/* eslint-disable */
+import { redirect } from "react-router-dom";
+import { deleteContact } from "../components/Contact";
+
+export async function action({ params }) {
+  await deleteContact(params.contactId);
+  return redirect("/");
+}
```
Comment (Ghiftee) [inline src/routes/destroy.jsx]: - [x] I do not think this file is related to this particular project, kindly remove it to make your project more professional.

File: src/routes/edit.jsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
+/* eslint-disable */
+import { Form, useLoaderData, redirect, useNavigate } from "react-router-dom";
+import { updateContact } from "../components/Contact";
+
+export async function action({ request, params }) {
+  const formData = await request.formData();
+  const updates = Object.fromEntries(formData);
+  await updateContact(params.contactId, updates);
+  return redirect(`/contacts/${params.contactId}`);
```
Comment (Ghiftee) [inline src/routes/edit.jsx]: - [x] I do not think this file is related to this particular project, kindly remove it to make your project more professional.

File: src/routes/index.jsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+		This is a demo for React Router.
+		<br />
+		Check out{" "}
+		<a href="https://reactrouter.com">
+				the docs at reactrouter.com
+		</a>
+		.
+		</p>
+	);
+}
```
Comment (Ghiftee) [inline src/routes/index.jsx]: - [x] I do not think this file is related to this particular project, kindly remove it to make your project more professional.

File: src/routes/root.jsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+export default function Root() {
+  const { contacts, q } = useLoaderData();
+  const navigation = useNavigation();
+  const submit = useSubmit();
+
+  const searching = navigation.location && new URLSearchParams(navigation.location.search).has('q');
+
+  // useEffect(() => {
+  //   document.getElementById("q").value = q;
+  // }, [q]);
```
Comment (Ghiftee) [inline src/routes/root.jsx]: - [x] Kindly remove commented code and let's try to ensure that only code related to this project is here

File: src/routes/contact.jsx [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+import { Form, useFetcher, useLoaderData } from 'react-router-dom';
+import { getContact, updateContact } from '../components/Contact';
+
+export async function loader ({ params }) {
+  const contact = await getContact(params.contactId);
+  if (!contact) {
+    throw new Response("", {
+      status: 404,
+      statusText: "Not Found",
+    });
```
Comment (Ghiftee) [inline src/routes/contact.jsx]: - [x] I do not think this file is related to this particular project, kindly remove it to make your project more professional.

File: src/redux/rockets/rocketSlice.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
+/* eslint-disable no-param-reassign */
```
Comment (Ghiftee) [inline src/redux/rockets/rocketSlice.js]: - [x] Let's try to not disable linters as they help us write cleaner code.

File: src/redux/missions/missionSlice.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
+/* eslint-disable no-param-reassign */
```
Comment (Ghiftee) [inline src/redux/missions/missionSlice.js]: - [x] Let's try to not disable linters as they help us write cleaner code.

File: src/index.css [hunk 8]
```diff
@@ -0,0 +1,379 @@
...
+#sidebar h1 {
+  font-size: 1rem;
+  font-weight: 500;
+  display: flex;
+  align-items: center;
+  margin: 0;
+  padding: 1rem 2rem;
+  border-top: 1px solid #e3e3e3;
+
+  /* order: 1; */
```
Comment (Ghiftee) [inline src/index.css]: - [x] Kindly remove commented code

File: src/error-page.jsx [hunk 9]
```diff
@@ -0,0 +1,17 @@
+/* eslint-disable */
```
Comment (Ghiftee) [inline src/error-page.jsx]: - [x] Let's try to not disable linters as they help us write cleaner code.

File: src/components/rockets/DisplayRocket.module.css [hunk 10]
```diff
@@ -0,0 +1,49 @@
+* {
+  /* background-color: brown; */
```
Comment (Ghiftee) [inline src/components/rockets/DisplayRocket.module.css]: - [x] Kindly remove commented code

File: src/components/mission/missions.css [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+  font-weight: 500;
+  font-size: 14px;
+  border: 1px solid;
+  border-radius: 4px;
+  padding: 8px;
+}
+
+@media screen and (max-width: 900px) {
+
+  /* stylelint-disable-next-line rule-empty-line-before */
```
Comment (Ghiftee) [inline src/components/mission/missions.css]: - [x] Let's try to not disable linters as they help us write cleaner code.

File: src/components/Contact.js [hunk 12]
```diff
@@ -0,0 +1,74 @@
+/* eslint-disable */
```
Comment (Ghiftee) [inline src/components/Contact.js]: - [x] I do not think this file is related to this particular project, kindly remove it to make your project more professional.

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hello team 👋🏽 

Good job working on the project so far! 
To highlight 👍🏽 
✔️ You added placeholder messages for when there are no missions and rockets
✔️ Your tests pass
✔️ Both partners gave reviews
However, there are some issues that you need to work on to go to the next project but you are almost there!



## Required Changes ♻️
_Check the comments under the review._

- [x] When I add a mission, it is added twice as shown below, kindly fix it
Screenshot 2023-01-07 at 08 54 34 (2)

- [x] Kindly add more unit tests to ensure all functions and components are tested.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by muneebulrehman — APPROVED** [review]
Hi team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🥳
Congratulations! 🎉 

# Status: Approved✔️

## Highlights
- No linter errors
- Descriptive PR
- Good UI

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**